apollo-federation/src/query_graph/graph_path.rs

```diff
@@ -1554,11 +1629,124 @@ where
         excluded_conditions: &ExcludedConditions,
     ) -> Result<ConditionResolution, FederationError> {
         let edge_weight = self.graph.edge_weight(edge)?;
-        if edge_weight.conditions.is_none() {
+        if edge_weight.conditions.is_none() && edge_weight.required_contexts.is_empty() {
             return Ok(ConditionResolution::no_conditions());
         }
+
+        /* Resolve context conditions */
+
+        let mut total_cost = 0.;
+        let mut context_map: IndexMap<String, ContextMapEntry> = IndexMap::default();
+
+        if !edge_weight.required_contexts.is_empty() {
+            let schema = self.graph.schema()?;
+            let mut was_unsatisfied = false;
+            for ctx in &edge_weight.required_contexts {
+                let mut levels_in_query_path = 0;
+                let mut levels_in_data_path = 0;
+                // TODO: Verify that `self.edges` and `self.edge_triggers` are the same length.
+                // Otherwise, this loop will be truncated.
```

Comment:
I assume this is the case, but I want to double-check.
```suggestion
```

Reply:
Yes, they will be the same length. See the comment in my PR about generating a `GraphPathItem` from these vectors.